src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelStateEntry.cs

```diff
@@ -23,7 +24,17 @@ public abstract class ModelStateEntry
         /// <summary>
         /// Gets the <see cref="ModelErrorCollection"/> for this entry.
         /// </summary>
-        public ModelErrorCollection Errors { get; } = new ModelErrorCollection();
+        public ModelErrorCollection Errors
+        {
+            get
+            {
+                if(_errors == null)
+                {
+                    _errors = new ModelErrorCollection();
+                }
+                return _errors;
+            }
+        }
```

Comment:
@DavidObando @rynowak and I talked offline and both need to see numbers demonstrating this change improves anything. If (as I expect due to numerous reads of this property) it doesn't help, need to change more in order to avoid these allocations.